packages/frontend/src/hooks/thirdPartyServices/useSentry.ts

```diff
@@ -0,0 +1,23 @@
+import { HealthState } from '@lightdash/common';
+import * as Sentry from '@sentry/react';
+import { Integrations } from '@sentry/tracing';
+import { useEffect, useState } from 'react';
+
+const useSentry = (sentryConfig: HealthState['sentry'] | undefined) => {
+    const [isSentryLoaded, setIsSentryLoaded] = useState(false);
+
+    useEffect(() => {
+        if (sentryConfig && !isSentryLoaded && sentryConfig.dsn) {
+            Sentry.init({
+                dsn: sentryConfig.dsn,
+                release: sentryConfig.release,
+                environment: sentryConfig.environment,
+                integrations: [new Integrations.BrowserTracing()],
+                tracesSampleRate: 1.0,
+            });
+            setIsSentryLoaded(true);
+        }
+    }, [isSentryLoaded, setIsSentryLoaded, sentryConfig]);
```

Comment:
this is definitely not relevant but recently I found out about https://partytown.builder.io/ which basically runs third party scripts to a webworker (improving overall performance) not sure about sentry but Cohere must be eating a lot of resources. do you think this is worth considering (separately from this PR of course)?

Reply:
Good idea 🎉 opened a ticket for that